### PR TITLE
fix: resolve the "eventTargets" error in JSDOM by dropping "instanceof" 

### LINKED ISSUE #1785: TypeError: The "eventTargets" argument must be an instance of EventEmitter or EventTarget
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 18 or higher

### Node.js version

v18

### Reproduction repository

https://github.com/mswjs/examples/actions/runs/6612009094/job/17956995430?pr=92

### Reproduction steps

```
pnpm test
```

### Current behavior

```
 Cause:
    TypeError: The "eventTargets" argument must be an instance of EventEmitter or EventTarget. Received an instance of AbortSignal
```

### Expected behavior

No errors are thrown.

### LINKED ISSUE #1803: Create React App / Jest: Axios requests aren’t intercepted by MSW
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 18 or higher

### Node.js version

v18.17.1

### Reproduction repository

https://github.com/joel-daros/msw2-text-encoder-issue/tree/main

### Reproduction steps

I created a simple repository with an example, testing `DogService.ts`:

```ts
import axios from "axios";

const axiosInstance = axios.create();

type DogFacts = {
  facts: string[];
};

export const getDogFacts = async (): Promise<string[]> => {
  const response = await axiosInstance.get<DogFacts>(
    "https://dogapi.dog/api/facts"
  );
  return response.data.facts;
};
```

The test:

```ts
import { setupServer } from "msw/node";
import { HttpResponse, http } from "msw";
import { getDogFacts } from "./DogService";

const server = setupServer();

beforeAll(() => server.listen());
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

server.events.on("request:start", ({ request }) => {
  console.log("MSW intercepted:", request.method, request.url);
});

it("returns the dog facts", async () => {
  server.use(
    http.get("/api/facts", () => {
      console.log("getDogFacts was intercepted by MSW");
      return HttpResponse.json(["fact1", "fact2"], { status: 200 });
    })
  );
  const result = await getDogFacts();
  expect(result).toStrictEqual(["fact1", "fact2"]);
});
```

But it looks like the MSW never intercepted the request, the test fails with the error:

```sh
 FAIL  src/DogService.test.ts
  ✕ returns the dog facts (33 ms)

  ● returns the dog facts

    AxiosError: Network Error

      at XMLHttpRequest.invoke (node_modules/axios/lib/adapters/xhr.js:163:14)
      at XMLHttpRequest.call (node_modules/@mswjs/interceptors/lib/node/chunk-JCWVLTP7.js:337:20)
      at Proxy.call (node_modules/@mswjs/interceptors/lib/node/chunk-JCWVLTP7.js:176:29)
      at XMLHttpRequestController.trigger (node_modules/@mswjs/interceptors/lib/node/chunk-JCWVLTP7.js:579:16)
      at XMLHttpRequestController.errorWith (node_modules/@mswjs/interceptors/lib/node/chunk-JCWVLTP7.js:547:10)
      at XMLHttpRequestController.xhrRequestController.onRequest (node_modules/@mswjs/interceptors/lib/node/chunk-JCWVLTP7.js:696:32)
```

### Current behavior

Test is failing

### Expected behavior

MSW intercept the request correctly returning the mocked response and test pass.

### PR DESCRIPTION
**Pull request by mattcosta7**
fixes https://github.com/mswjs/msw/issues/1785
fixes https://github.com/mswjs/msw/issues/1803 (I think)

Avoid using instanceof to deal with jest/jsdom potentially overriding the global error 

I haven't had a chance to try and validate this assumption, but I am assuming that this check is failing on the `instanceof Error` and therefore rethrowing instead of being caught.

In the past I've seen this with jest overriding globals such that these instanceof checks don't always do what's expected

I validated this locally by pulling `mswjs/examples` and modifying the installed version in the `with-jest` `node_modules`. Once this was done the tests in that repo pass correctly with `msw@2+` whereas without this change they fail

### CODE REVIEW

**Comment by pord911** [comment]
@mattcosta7 
I can roughly confirm your theory. See highlighted local variable value.

image

Happens when TypeError message is thrown within the following environment:

jest 27
jsdom 16
node 18
msw 2.0.0

Found [this](https://github.com/jsdom/jsdom/blob/a39e0ec4ce9a8806692d986a7ed0cd565ec7498a/Changelog.md?plain=1#L137) on jsdom changelog, even though it is for version 18 of jsdom, but here (in my environemnt) it seems to be the same case.

File: src/node/utils/isNodeExceptionLike.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * Determines if the given value is shaped like a Node.js exception.
+ * Node.js exceptions have additional information, like
+ * the `code` and `errno` properties.
+ *
+ * In some environments, particularly jsdom/jest these may not
+ * instances of `Error` or its subclasses, despite being similar
+ * to them.
+ */
+export function isNodeExceptionLike(
```
Comment (mattcosta7) [inline src/node/utils/isNodeExceptionLike.ts:10]: renamed this function and added some docs that it's _very similar_ to a `NodeException` but isn't guaranteed to be and `instanceof` it

**Comment by kettanaito** [comment]
Thanks for fixing this, @mattcosta7. Yet another day, yet another JSDOM shenanigans. I will give this a look and let's have this merged.

**Review by kettanaito — APPROVED** [review]
Looks great.

**Comment by sauldeleon** [comment]
Thanks @mattcosta7  @kettanaito !!

**Comment by kettanaito** [comment]
## Released: v2.0.2 🎉

This has been released in v2.0.2!

- 📄 [**Release notes**](https://github.com/mswjs/msw/releases/tag/v2.0.2)
- 📦 [npm package](https://www.npmjs.com/package/msw/v/2.0.2)

Make sure to always update to the latest version (`npm i msw@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.

**Pull request merged**